wrap/gtwrap/matlab_wrapper/mixins.py

```diff
@@ -0,0 +1,222 @@
+"""Mixins for reducing the amount of boilerplate in the main wrapper class."""
+
+import gtwrap.interface_parser as parser
+import gtwrap.template_instantiator as instantiator
+
+
+class CheckMixin:
+    """Mixin to provide various checks."""
+    # Data types that are primitive types
+    not_ptr_type = ['int', 'double', 'bool', 'char', 'unsigned char', 'size_t']
+    # Ignore the namespace for these datatypes
+    ignore_namespace = ['Matrix', 'Vector', 'Point2', 'Point3']
+    # Methods that should be ignored
+    ignore_methods = ['pickle']
+    # Methods that should not be wrapped directly
+    whitelist = ['serializable', 'serialize']
+    # Datatypes that do not need to be checked in methods
+    not_check_type: list = []
+
+    def _has_serialization(self, cls):
+        for m in cls.methods:
+            if m.name in self.whitelist:
+                return True
+        return False
+
+    def is_shared_ptr(self, arg_type):
```

Comment:
are we missing type hints in this file?

Reply:
Ah shoot. Updating the matlab wrapper was such an onerous task that I didn't think to add type-hints. That can be a future PR since it is not quite related to this functionality.